### PR TITLE
[Proof] Change AccumulatorConsistencyProof to use the new format

### PR DESCRIPTION
**Pull request by wqfish**
## Motivation

In https://github.com/libra/libra/pull/828 we decided to use this new format. Now change the code accordingly.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y.

## Test Plan

CI.

## Related PRs

None.

### CODE REVIEW

**Comment by msmouse** [comment]
LGTM
@bors-libra delegate+

File: types/src/proof/definition.rs [hunk 1]
```diff
@@ -252,92 +252,40 @@ impl IntoProto for SparseMerkleProof {
...
 }
 
 impl FromProto for AccumulatorConsistencyProof {
     type ProtoType = crate::proto::proof::AccumulatorConsistencyProof;
 
     fn from_proto(mut proto_proof: Self::ProtoType) -> Result<Self> {
-        let frozen_subtree_roots = proto_proof
-            .take_frozen_subtree_roots()
+        let subtrees = proto_proof
+            .take_subtrees()
```
Comment (zamsden) [inline types/src/proof/definition.rs]: Where is take_subtrees defined?  Is there a missing commit in this diff?

File: types/src/proto/proof.proto [hunk 2]
```diff
@@ -52,18 +52,9 @@ message SparseMerkleProof {
...
-  uint32 num_siblings = 2;
-
-  // The non-default siblings. Note that the entire list of siblings always
-  // start of zero or more non-default siblings, followed by zero of more
-  // default siblings. So given the total number of siblings and the non-default
-  // siblings we should be able to construct the entire sibling list.
-  repeated bytes non_default_siblings = 3;
+  // The root hashes of the subtrees that represent new leaves. Note that none
+  // of these hashes should be default hash.
+  repeated bytes subtrees = 1;
```
Comment (zamsden) [inline types/src/proto/proof.proto]: OIC

**Pull request closed without merging**